services/app-graphql/src/schema.graphql

```diff
@@ -577,6 +642,15 @@ type UpdateCMSUserPayload {
     user: CMSUser!
 }
 
+input UpdateContractDraftRevisionInput {
+    contractID: ID!
+    draftRevision: ContractDraftRevisionInput!
```

Comment:
I would not let you send a whole draft revision here. State users are only allowed to change the FormData, they can't change updatedAt or submittedAt or the package name or any of that, so the only thing they should be allowed to send is the form data. 

That's how we handle it with updateDraftContractRates and with submitRate. In general it's best to avoid API inputs that are just going to be ignored, or worse, untrusted.

Reply:
Good call, there is a lot of data we don't use in the revision. I added the entire revision just to get `updatedAt` so we can check for concurrent editing. My other thought was to just send `updatedAt`, but that seemed vague on where this field came from. Another thought is a new graphQL type for meta data for inputs that includes these types of fields, so its better defined. Also, probably all this data should not be in the `input` variable.

```graphql
input UpdateContractDraftRevisionInput {
    contractID: ID!
    updatedAt: DateTime!
    formData: ContractDraftRevisionFormDataInput!
}
```

I took a look at `updateDraftContractRates` and noticed we don't check for concurrent updates there. I was able to overwrite new rate data with stale rate data using two windows of the app.

Reply:
I like just passing in updatedAt, I would maybe call it `lastSeenUpdatedAt` to be more explicit?

Reply:
@macrael I'm going to make a new PR to add checking for concurrent update in `updateDraftContractRates`